### PR TITLE
prepend rank number to all print() invocations

### PR DESCRIPTION
**Pull request by matthiasdiener**
**Questions for the review**:
- [ ] Is the scope and purpose of the PR clear?
  - [ ] The PR should have a description.
  - [ ] The PR should have a guide if needed (e.g., an ordering).
- [ ] Is every top-level method and class documented? Are things that should be documented actually so?
- [ ] Is the interface understandable? (I.e. can someone figure out what stuff does?) Is it well-defined?
- [ ] Does the implementation do what the docstring claims?
- [ ] Is everything that is implemented covered by tests?
- [ ] Do you see any immediate risks or performance disadvantages with the design? Example: what do interface normals attach to?

### CODE REVIEW

**Review by inducer — CHANGES_REQUESTED** [review]
Thanks for working on this, but sorry, I'm not a fan. Changing builtins is kind of a no-no IMO. A better approach might be to use logging more pervasively and then install a logging formatter that automatically includes the rank.

**Comment by matthiasdiener** [comment]
> Thanks for working on this, but sorry, I'm not a fan. Changing builtins is kind of a no-no IMO. A better approach might be to use logging more pervasively and then install a logging formatter that automatically includes the rank.

I was looking at logging formatters as well but didn't find a solution that propagates the formatting across packages; i.e., we can add a formatter to mirgecom logging, but that won't affect grudge etc.

**Comment by MTCam** [comment]
> > Thanks for working on this, but sorry, I'm not a fan. Changing builtins is kind of a no-no IMO. A better approach might be to use logging more pervasively and then install a logging formatter that automatically includes the rank.
> 
> I was looking at logging formatters as well but didn't find a solution that propagates the formatting across packages; i.e., we can add a formatter to mirgecom logging, but that won't affect logging in grudge etc.

It is a miserable situation to need to debug at the level where you need to see the rank of the writer, but I agree that this is extremely useful when one needs.   If we can turn it on-and-off, it seems only slightly sinful to tweak the internal function.

**Comment by matthiasdiener** [comment]
Maybe it would be enough to wrap this in an `if 0:` so that it must be manually enabled?

**Comment by inducer** [comment]
I'm OK with this if it's not turned on unconditionally.

**Comment by matthiasdiener** [comment]
> I'm OK with this if it's not turned on unconditionally (i.e. on import).

What do you think of 5915121?

**Review by MTCam — APPROVED** [review]
Looks useful.

File: mirgecom/mpi.py [hunk 1]
```diff
@@ -184,3 +184,21 @@ def wrapped_func(*args: Any, **kwargs: Any):
         debug_remote_on_single_rank(MPI.COMM_WORLD, 0, func, *args, **kwargs)
 
     return wrapped_func
+
+
+# Change this to `if 1:` in order to prepend the rank number to every message
+# printed with print().
+if 0:
```
Comment (inducer) [inline mirgecom/mpi.py:191]:
```suggestion
def enable_rank_labeled_print():
    """Add docstring here.
    """
```
(and hook into docs)

**Review by inducer — APPROVED** [review]

File: mirgecom/mpi.py (hunk 1 above)
Comment (matthiasdiener) [reply mirgecom/mpi.py]: Done, thanks!

Comment (MTCam) [reply mirgecom/mpi.py]: Even better. This is cool.

**Pull request merged**